Publishes MinVer.Cli as Dotnet Tool Package

thefringeninja opened the pull request:
Unfortunatley I cannot build this on ubuntu at the moment due to libgit2/libgit2sharp#1626.

adamralph commented on targets/Program.cs:
```diff
@@ -27,7 +27,11 @@ public static Task Main(string[] args)
+            {
+                await RunAsync("dotnet", $"pack ./MinVer/MinVer.csproj --configuration Release --no-build");
+                await RunAsync("dotnet", $"pack ./MinVer.Cli/MinVer.Cli.csproj --configuration Release --no-build");
+            });
```
FYI this can be a single statement with a `forEach` input.

bording commented:
@thefringeninja Which version of Ubuntu are you trying?

thefringeninja commented:
in fact it was at build time
```
Bullseye: Starting... (default)
Bullseye/build: Starting...
dotnet build MinVer.sln --configuration Release
Microsoft (R) Build Engine version 15.8.169+g1ccb72aefa for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 41.44 ms for /home/jwow/dev/min-ver/MinVer/MinVer.csproj.
  Restore completed in 41.44 ms for /home/jwow/dev/min-ver/MinVer.Cli/MinVer.Cli.csproj.
  Restore completed in 41.44 ms for /home/jwow/dev/min-ver/targets/Targets.csproj.
  Restore completed in 41.44 ms for /home/jwow/dev/min-ver/MinVerTests/MinVerTests.csproj.
  Targets -> /home/jwow/dev/min-ver/targets/bin/Release/netcoreapp2.1/Targets.dll
/home/jwow/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: The "Microsoft.Build.Tasks.Git.LocateRepository" task failed unexpectedly. [/home/jwow/dev/min-ver/MinVer/MinVer.csproj]
/home/jwow/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load shared library '/home/jwow/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/../tools/netcoreapp2.0/runtimes/linux-x64/native/libgit2-b0d9952.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/jwow/.nuget/packages/microsoft.build.tasks.git/1.0.0-beta-63127-02/build/../tools/netcoreapp2.0/runtimes/linux-x64/native/libgit2-b0d9952.so) [/home/jwow/dev/min-ver/MinVer/MinVer.csproj]
```
as for dotnet info:
```
jwow@ronald-drumpf  ~/dev/min-ver   dotnet-tool  dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.403
 Commit:    04e15494b6

Runtime Environment:
 OS Name:     linuxmint
 OS Version:  19
 OS Platform: Linux
 RID:         linuxmint.19-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.403/

Host (useful for support):
  Version: 2.1.5
  Commit:  290303f510

.NET Core SDKs installed:
  2.1.403 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

thefringeninja commented:
The good news is I think I have a workaround. I added this PPA: https://launchpad.net/~xapienz/+archive/ubuntu/curl34 and updated my curl package from there. It is building correctly now! Perhaps a note in the README would help out others here.

bording commented:
> in fact it was at build time

Oh, okay that is because of the SourceLink package, not the LibGit2Sharp usage in this project. That makes more sense.

There is a new preview release of LibGit2Sharp that includes a binary that should work on on Ubuntu 18.04 / Mint 19 (since your output shows the RID as `linuxmint.19-x64`), but the SourceLink package hasn't been updated to use it yet.
